### PR TITLE
replace code level class symbol with inferface

### PR DESCRIPTION
**Pull request by zjhmale**
relates to #3375

### CODE REVIEW

File: src/Idris/AbsSyntaxTree.hs [hunk 1]
```diff
@@ -1412,18 +1412,18 @@ highestFC (PAppImpl t _)          = highestFC t
...
-  , class_defaults             :: [(Name, (Name, PDecl))]         -- ^ method name -> default impl
-  , class_default_superclasses :: [PDecl]
-  , class_params               :: [Name]
-  , class_instances            :: [(Name, Bool)] -- ^ the Bool is whether to include in instance search, so named instances are excluded
-  , class_determiners          :: [Int]
+data InterfaceInfo = CI {
+    instanceCtorName               :: Name
+  , interface_methods              :: [(Name, (Bool, FnOpts, PTerm))] -- ^ flag whether it's a "data" method
+  , interface_defaults             :: [(Name, (Name, PDecl))]         -- ^ method name -> default impl
+  , class_default_super_interfaces :: [PDecl]
```
Comment (Melvar) [inline src/Idris/AbsSyntaxTree.hs]: Probably wanted to replace both instances of “class” in this field name.

Comment (zjhmale) [inline src/Idris/AbsSyntaxTree.hs]: yeah, i just missed it 😅

**Comment by melted** [comment]
The renaming Idris.Eval.Class to Idris.Eval.Interface must be done in idris.cabal too.

This is why Travis is failing.

**Comment by zjhmale** [comment]
😅 forgot about the cabal file
>The renaming Idris.Eval.Class to Idris.Eval.Interface must be done in idris.cabal too.
This is why Travis is failing.

**Comment by melted** [comment]
Thanks for the PR!

**Pull request merged**